metaflow/plugins/aws/utils.py

```diff
@@ -0,0 +1,160 @@
+            tar.add(DATASTORE_LOCAL_DIR)
+        # At this point we store it in the datastore; we
+        # save it as raw data in the flow datastore and save a pointer
+        # to it as metadata for the task
+        with open(tar_file_path, 'rb') as f:
+            _, key = task_ds.parent_datastore.save_data([f])[0]
+        task_ds.save_metadata({'local_metadata': key})
+
+
+def sync_local_metadata_from_datastore(metadata, task_datastore):
+    # Do nothing if metadata is not local
+    if metadata.TYPE != 'local':
+        return
+
+    # Otherwise, copy metadata from task datastore to local metadata store
+    def echo_none(*args, **kwargs):
+        pass
+    key_to_load = task_datastore.load_metadata(
+        ['local_metadata'])['local_metadata']
+    tarball = task_datastore.parent_datastore.load_data(
+        [key_to_load])[0]
+    with util.TempDir() as td:
+        with tarfile.open(fileobj=BytesIO(tarball), mode='r:gz') as tar:
+            tar.extractall(td)
+        copy_tree(
+            os.path.join(td, DATASTORE_LOCAL_DIR),
+            LocalBackend.get_datastore_root_from_config(echo_none),
+            update=True)
+
+
+class CommonTaskAttrs:
+    def __init__(self,
+        flow_name,
+        step_name,
+        run_id,
+        task_id,
+        attempt,
+        user,
+        version,
+    ):
+        self.flow_name = flow_name
+        self.step_name = step_name
+        self.run_id = run_id
+        self.task_id = task_id
+        self.attempt = attempt
+        self.user = user
+        self.version = version
+
+    def to_dict(self, key_prefix):
+        attrs = {
+            key_prefix + 'flow_name': self.flow_name,
+            key_prefix + 'step_name': self.step_name,
+            key_prefix + 'run_id': self.run_id,
+            key_prefix + 'task_id': self.task_id,
+            key_prefix + 'retry_count': str(self.attempt),
+            key_prefix + 'version': self.version,
+        }
+        if self.user is not None:
+            attrs[key_prefix + 'user'] = self.user
+        return attrs
+
+def get_writeable_datastore(
+    common_task_attrs,
+    datastore_class,
+):
+    return datastore_class(
+            mode='w',
+            flow_name=common_task_attrs.flow_name,
+            step_name=common_task_attrs.step_name,
+            run_id=common_task_attrs.run_id,
+            task_id=common_task_attrs.task_id,
+            attempt=common_task_attrs.attempt
+    )
+
+def get_datastore_root(
+    common_task_attrs,
+    datastore_class,
+):
+    return datastore_class.make_path(common_task_attrs.flow_name, common_task_attrs.run_id, common_task_attrs.step_name, common_task_attrs.task_id)
+
+
+def build_task_command(
+    environment,
+    code_package_url,
+    step_name,
+    step_cmds,
+    flow_name,
+    run_id,
+    task_id,
+    attempt,
+    stdout_path,
+    stderr_path,
+):
+    mflog_expr = export_mflog_env_vars(datastore_type='s3',
```

Comment:
nit: the variables listed here are not in the order specified by the call making it slightly more confusing.